File: src/matlab/com/springbok/pattern/PatternENST806V01Test.m
classdef PatternENST806V01Test < TestUtility
% Tests methods of PatternENST806V01 class.
  
% Copyright (C) 2016 Springbok LLC <http://springbok.io/> All rights reserved.

  properties (Constant = true)
    
    GainMax_input_1 = 54.4;
    Efficiency_input = 0.7;
    CoefA_input = 35;
    
    self_1 = struct( ...
        'GainMax', 54.4, ...
        'Efficiency', 0.7, ...
        'CoefA', 35, ...
        'phi_m', 0.441194158165796, ...
        'phi_r', 1.000000000000000, ...
        'phi_b', 63.095734448019329, ...
        'd_over_lambda', 199.664616023944350, ...
        'G1', 35.000000000000000);
    
    GainMax_input_2 = 34.4;
    
    self_2 = struct( ...
        'GainMax', 34.4, ...
        'Efficiency', 0.7, ...
        'CoefA', 35, ...
        'phi_m', 4.116949087625147, ...
        'phi_r', 5.008398683320418, ...
        'phi_b', 63.095734448019329, ...
        'd_over_lambda', 19.966461602394443, ...
        'G1', 17.507527682468442);
    
    Phi_input = [0.2; 0.7; 4.5; 20; 150];
    
    G_expected_1 = [ ...
        50.413404110801082
        35.000000000000000
        18.669687155616412
        2.474250108400469
        -10.000000000000000
                   ];
    Gx_expected_1 = [ ...
        50.413404110801082
        35.000000000000000
        18.669687155616412
        2.474250108400469
        -10.000000000000000
                    ];
    
    G_expected_2 = [ ...
        34.360134041108012
        33.911642003573128
        17.507527682468442
        2.474250108400469
        -10.000000000000000
                   ];
    Gx_expected_2 = [ ...
        34.360134041108012
        33.911642003573128
        17.507527682468442
        2.474250108400469
        -10.000000000000000
                    ];
    
  end % properties (Constant = true)
  
  methods
    
    function this = PatternENST806V01Test(logFId, testMode)
    % Constructs a PatternENST806V01Test
    %
    % Parameters:
    %   logFId - Log file identifier
    %   testMode - Test mode, if 'interactive' then beeps and pauses
      
    % Invoke superclass constructor
      if nargin == 0
        superArgs = {};
        
      else
        superArgs{1} = logFId;
        superArgs{2} = testMode;
        
      end % if
      this@TestUtility(superArgs{:});
      
    end % PatternENST806V01Test()
    
    function test_PatternENST806V01(this)
    % Tests PatternENST806V01 method.
      
      p_1 = PatternENST806V01(this.GainMax_input_1, this.Efficiency_input, this.CoefA_input);
      
      this.assert_true( ...
          'PatternENST806V01', ...
          sprintf('PatternENST806V01:%.1f', this.GainMax_input_1), ...
          this.HIGH_PRECISION_DESC, ...
          PatternUtility.compare_pattern(this.self_1, p_1));
      
      p_2 = PatternENST806V01(this.GainMax_input_2, this.Efficiency_input, this.CoefA_input);
      
      this.assert_true( ...
          'PatternENST806V01', ...
          sprintf('PatternENST806V01:%.1f', this.GainMax_input_2), ...
          this.HIGH_PRECISION_DESC, ...
          PatternUtility.compare_pattern(this.self_2, p_2));
      
    end % test_PatternENST806V01()
    
    function test_gain(this)
    % Tests gain method.
      
      warning('off');
      
      p_1 = PatternENST806V01(this.GainMax_input_1, this.Efficiency_input, this.CoefA_input);
      
      [G_actual, Gx_actual] = p_1.gain(this.Phi_input);
      
      t(1) = max(abs(G_actual - this.G_expected_1) < this.HIGH_PRECISION);
      t(2) = max(abs(Gx_actual - this.Gx_expected_1) < this.HIGH_PRECISION);
      
      this.assert_true( ...
          'PatternENST806V01', ...
          sprintf('gain:%.1f', this.GainMax_input_1), ...
          this.HIGH_PRECISION_DESC, ...
          min(t));
      
      [G, Gx, fH] = p_1.gain([0:.001:180], 'PlotFlag', true);
      pause(1); close(fH);
      
      p_2 = PatternENST806V01(this.GainMax_input_2, this.Efficiency_input, this.CoefA_input);
      
      [G_actual, Gx_actual] = p_2.gain(this.Phi_input);
      
      t(1) = max(abs(G_actual - this.G_expected_2) < this.HIGH_PRECISION);
      t(2) = max(abs(Gx_actual - this.Gx_expected_2) < this.HIGH_PRECISION);
      
      this.assert_true( ...
          'PatternENST806V01', ...
          sprintf('gain:%.1f', this.GainMax_input_2), ...
          this.HIGH_PRECISION_DESC, ...
          min(t));
      
      [G, Gx, fH] = p_2.gain([0:.001:180], 'PlotFlag', true);
      pause(1); close(fH);
      
      warning('on');
      
    end % test_gain()
    
  end % methods
  
end % classdef

File: src/java/com/springbok/pattern/PatternENST806V01Test.java
/* Copyright (C) 2016 Springbok LLC <http://springbok.io/> All rights reserved. */

package com.springbok.pattern;

import com.springbok.utility.TestUtility;
import org.junit.Test;

import java.util.HashMap;
import java.util.Map;

import static org.junit.Assert.assertTrue;

/**
 * Tests methods of PatternENST806V01 class.
 */
public class PatternENST806V01Test {
	private static final double GainMax_input_1 = 54.4;
	private static final double Efficiency_input = 0.7;
	private static final double CoefA_input = 35.0;

	private static final double GainMax_input_2 = 34.4;

	private static final double[] Phi_input = new double[] { 0.2, 0.7, 4.5, 20.0, 150.0 };
	private static final double[] G_expected_1 = new double[] { 50.413404110801082, 35.0, 18.669687155616412,
			2.474250108400469, -10.0 };
	private static final double[] Gx_expected_1 = new double[] { 50.413404110801082, 35.0, 18.669687155616412,
			2.474250108400469, -10.0 };
	private static final double[] G_expected_2 = new double[] { 34.360134041108012, 33.911642003573128,
			17.507527682468442, 2.474250108400469, -10.0 };
	private static final double[] Gx_expected_2 = new double[] { 34.360134041108012, 33.911642003573128,
			17.507527682468442, 2.474250108400469, -10.0 };

	/**
	 * Tests PatternENST806V01 method.
	 */
	@Test
	public void test_PatternENST806V01() {
		PatternENST806V01 p_1 = new PatternENST806V01(GainMax_input_1, Efficiency_input, CoefA_input);

		assertTrue(TestUtility.isDoublesEquals(54.4, p_1.getGainMax()));
		assertTrue(TestUtility.isDoublesEquals(0.7, p_1.getEfficiency()));
		assertTrue(TestUtility.isDoublesEquals(35, p_1.getCoefA()));
		assertTrue(TestUtility.isDoublesEquals(0.441194158165796, p_1.getPhi_m()));
		assertTrue(TestUtility.isDoublesEquals(1.0, p_1.getPhi_r()));
		assertTrue(TestUtility.isDoublesEquals(63.095734448019329, p_1.getPhi_b()));
		assertTrue(TestUtility.isDoublesEquals(199.664616023944350, p_1.getD_over_lambda()));
		assertTrue(TestUtility.isDoublesEquals(35.0, p_1.getG1()));

		PatternENST806V01 p_2 = new PatternENST806V01(GainMax_input_2, Efficiency_input, CoefA_input);

		assertTrue(TestUtility.isDoublesEquals(34.4, p_2.getGainMax()));
		assertTrue(TestUtility.isDoublesEquals(0.7, p_2.getEfficiency()));
		assertTrue(TestUtility.isDoublesEquals(35, p_2.getCoefA()));
		assertTrue(TestUtility.isDoublesEquals(4.116949087625147, p_2.getPhi_m()));
		assertTrue(TestUtility.isDoublesEquals(5.008398683320418, p_2.getPhi_r()));
		assertTrue(TestUtility.isDoublesEquals(63.095734448019329, p_2.getPhi_b()));
		assertTrue(TestUtility.isDoublesEquals(19.966461602394443, p_2.getD_over_lambda()));
		assertTrue(TestUtility.isDoublesEquals(17.507527682468442, p_2.getG1()));
	}

	/**
	 * Tests gain method.
	 */
	@Test
	public void test_gain() {
		PatternENST806V01 p_1 = new PatternENST806V01(GainMax_input_1, Efficiency_input, CoefA_input);

		Gain[] gains = new Gain[Phi_input.length];
		Map<String, Object> options = new HashMap<String, Object>();
		options.put("DoValidate", false);

		for (int i = 0; i < gains.length; i++) {
			Gain g = p_1.gain(Phi_input[i], options);
			gains[i] = g;
		}
		for (int i = 0; i < gains.length; i++) {
			assertTrue(TestUtility.isDoublesEquals(G_expected_1[i], gains[i].G));
			assertTrue(TestUtility.isDoublesEquals(Gx_expected_1[i], gains[i].Gx));
		}

		PatternENST806V01 p_2 = new PatternENST806V01(GainMax_input_2, Efficiency_input, CoefA_input);

		gains = new Gain[Phi_input.length];

		for (int i = 0; i < gains.length; i++) {
			Gain g = p_2.gain(Phi_input[i], options);
			gains[i] = g;
		}
		for (int i = 0; i < gains.length; i++) {
			assertTrue(TestUtility.isDoublesEquals(G_expected_2[i], gains[i].G));
			assertTrue(TestUtility.isDoublesEquals(Gx_expected_2[i], gains[i].Gx));
		}
	}
}
